src/lang/parser/lexer.ts

```diff
@@ -216,23 +230,22 @@ export class Lexer {
 					state.advance();
 					op += "=";
 				}
-				if (op !== "!") {
-					if (where) state.set(where);
-					else state.skip();
-					tok = { type: "operator", value: op, code: op, line: line };
-				}
 			}
-			if (tok === null) {
-				let type: any = null;
+
+			if (op && op !== "!") {
+				tok = { type: "operator", value: op, code: op, line: line };
+			} else {
```

Comment:
This restructuring was necessary so that TypeScript knows that tok is definitely assigned after the else branch ("all the rest").